presto-kudu/src/test/java/io/prestosql/plugin/kudu/TestKuduIntegrationDynamicFilter.java

```diff
@@ -0,0 +1,323 @@
+import io.prestosql.testing.ResultWithQueryId;
+import io.prestosql.tpch.TpchTable;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static io.airlift.testing.Assertions.assertGreaterThan;
+import static io.airlift.testing.Assertions.assertGreaterThanOrEqual;
+import static io.prestosql.SystemSessionProperties.ENABLE_LARGE_DYNAMIC_FILTERS;
+import static io.prestosql.SystemSessionProperties.JOIN_DISTRIBUTION_TYPE;
+import static io.prestosql.SystemSessionProperties.JOIN_REORDERING_STRATEGY;
+import static io.prestosql.plugin.kudu.KuduQueryRunnerFactory.createKuduQueryRunnerTpch;
+import static io.prestosql.spi.predicate.Domain.none;
+import static io.prestosql.spi.predicate.Domain.singleValue;
+import static io.prestosql.spi.predicate.Range.range;
+import static io.prestosql.spi.type.IntegerType.INTEGER;
+import static io.prestosql.sql.analyzer.FeaturesConfig.JoinDistributionType.PARTITIONED;
+import static io.prestosql.sql.analyzer.FeaturesConfig.JoinReorderingStrategy.NONE;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+public class TestKuduIntegrationDynamicFilter
+        extends AbstractTestQueryFramework
+{
+    private static final String DYNAMIC_FILTERING_WAIT_TIMEOUT = "dynamic_filtering_wait_timeout";
+    private static final long LINEITEM_COUNT = 60175;
+    private TestingKuduServer kuduServer;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        kuduServer = new TestingKuduServer();
+        return createKuduQueryRunnerTpch(
+            kuduServer, Optional.of(""),
+            ImmutableMap.of(JOIN_REORDERING_STRATEGY, NONE.name(),
+                JOIN_DISTRIBUTION_TYPE, PARTITIONED.name(),
+                ENABLE_LARGE_DYNAMIC_FILTERS, "true"),
+            ImmutableMap.of(DYNAMIC_FILTERING_WAIT_TIMEOUT, "1h"),
+            ImmutableMap.of("dynamic-filtering.large-partitioned.max-distinct-values-per-driver", "100",
+                "dynamic-filtering.large-partitioned.range-row-limit-per-driver", "100000"),
+            TpchTable.getTables());
+    }
+
+    @AfterClass(alwaysRun = true)
+    public final void destroy()
+    {
+        kuduServer.close();
+    }
+
+    @Test(timeOut = 30_000)
+    public void testJoinWithEmptyBuildSide()
+    {
+        DistributedQueryRunner runner = (DistributedQueryRunner) getQueryRunner();
+        ResultWithQueryId<MaterializedResult> result = runner.executeWithQueryId(
+                getSession(),
+                "SELECT * FROM lineitem JOIN supplier ON lineitem.suppkey = supplier.suppkey AND supplier.name = 'abc'");
+        assertEquals(result.getResult().getRowCount(), 0);
+
+        // TODO bring back OperatorStats assertions from https://github.com/prestosql/presto/commit/1feaa0f928a02f577c8ac9ef6cc0c8ec2008a46d
```

Comment:
Please check if adding assertions along the lines of `assertDynamicFiltering` in TestMemorySmoke works

Reply:
Added in the second commit

Reply:
Sorry for the misunderstanding, I actually meant adding assertions about rows read from Kudu in the tests here similar to the way `assertDynamicFiltering` works in TestMemorySmoke.

Reply:
Got it